src/Compilers/VisualBasic/Portable/Binding/TemporaryLookupResults.vb

```diff
@@ -0,0 +1,55 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.PooledObjects
+
+Namespace Microsoft.CodeAnalysis.VisualBasic
+    ''' <summary>
+    ''' TemporaryLookupResults allows the binding process to create an uncontested pool of 
+    ''' LookupResults that can be used by a single thread during binding.  LookupResults are
+    ''' retrieved/returned to this pool as needed, removing the need to get them from the
+    ''' global pool which may be highly contested on high-proc machines.
+    ''' <para/>
+    ''' TemporaryLookupResults instances can be created effectively as high up in the binding stack
+    ''' as is convenient.  Ideally, they are  created at any public entry-point into the binding
+    ''' process and passed along to any code that needs them.
+    ''' </summary>
+    Friend Structure TemporaryLookupResults
+        Private Shared ReadOnly s_stackPool As New ObjectPool(Of Stack(Of LookupResult))(Function() New Stack(Of LookupResult))
+
+        Private _stack As Stack(Of LookupResult)
+
+        ''' <summary>
+        ''' Constructor parameter is unused as exists solely to bypass the public default
+        ''' struct constructor.
+        ''' </summary>
+        Public Sub New(unused As Boolean)
```

Comment:
>Public Sub New(unused As Boolean) [](start = 8, length = 33)

It is not clear why this constructor is public. We could provide shared GetInstance() method instead that would use this private constructor.

Reply:
And instead of taking fake argument, it can actually take ```Stack(Of LookupResult)``` to use. 

---
In reply to: [361527158](https://github.com/dotnet/roslyn/pull/40522#discussion_r361527158) [](ancestors = 361527158)